libkbfs/prefetcher.go

```diff
@@ -425,6 +451,68 @@ func (p *blockPrefetcher) handlePrefetch(pre *prefetch, isPrefetchNew,
 	return numBlocks, isTail, nil
 }
 
+func (p *blockPrefetcher) rescheduleTopBlock(
+	blockID kbfsblock.ID, pp *prefetch, req *prefetchRequest) {
+	if len(pp.parents) > 0 {
+		p.cancelPrefetch(blockID, pp)
+		return
+	}
+
+	pre := p.prefetches[blockID]
+	delete(p.prefetches, blockID)
+	pp.Close()
+
+	// Only reschedule the top-most block, which has no parents.
+	rp, ok := p.rescheduled[blockID]
+	if !ok {
+		rp = &rescheduledPrefetch{
+			off: p.makeNewBackOff(),
+		}
+		p.rescheduled[blockID] = rp
+	}
+
+	if rp.timer != nil {
+		// Prefetch already scheduled.
+		return
+	}
+	req.ptr = BlockPointer{ID: blockID}
+	req.block = pre.req.block.NewEmpty()
```

Comment:
if you use `pre.req` here, then the `ptr` should already be correct, and you can instantiate a new block in place: `pre.req.block = pre.req.block.NewEmpty()`. Then you can pass `pre.req` to the `triggerPrefetch` below, and avoid passing `req` into this method at all.

Reply:
As below, I'm not sure if `pre.req` has all the right parameters in it.  If we can assume it does, that would be great!